### PR TITLE
Make vulkan build as optional

### PR DESCRIPTION
**Pull request by alexkalmuk**
Add cmake option `WITH_VULKAN`, which enables/disables Vulkan in Orbit in a straightforward way. It should introduce a bit of additional modularity to the Orbit. By default, this option set to `ON` (i.e., Vulkan enabled). 
This change affects only Linux, related to #4814.

I'm not sure about one place:
https://github.com/google/orbit/blob/3034421f88bab958f935cd3291f681870e1c9bad/src/OrbitGl/GpuSubmissionTrack.cpp#L61-L63
Basically, this file is compiled fine. On the other hand, it looks like it implicitly relies on Vulkan as a background layer.

### CODE REVIEW

File: src/LinuxTracingIntegrationTests/IntegrationTestPuppet.cpp [hunk 1]
```diff
@@ -194,7 +198,11 @@ int IntegrationTestPuppetMain() {
     } else if (command == PuppetConstants::kDlopenCommand) {
       LoadSoWithDlopenAndCallFunction();
     } else if (command == PuppetConstants::kVulkanTutorialCommand) {
+#ifdef VULKAN_ENABLED
       RunVulkanTutorial();
+#else
+      ORBIT_ERROR("Vulkan isn't enabled");
```
Comment (florian-kuebler) [inline src/LinuxTracingIntegrationTests/IntegrationTestPuppet.cpp:204]: Nit: Can you extend the comment to include the flag? Something like: "Compile with WITH_VULKAN".

Comment (alexkalmuk) [reply src/LinuxTracingIntegrationTests/IntegrationTestPuppet.cpp]: Thanks, done.

**Comment by florian-kuebler** [comment]
> By the way, I didn't update conanfile.py with the new option, which means that for Conan build Vulkan is enabled unconditionally.

I guess this is fine for now.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!
